api/src/services/attachment-service.ts

```diff
@@ -126,6 +133,14 @@ export class AttachmentService extends DBService {
       throw new HTTP400('Failed to delete project attachments record');
     }
 
-    await Promise.all(attachments.rows.map((attachment) => attachment.key).map(deleteFileFromS3));
+    for (const row of attachments.rows) {
+      await deleteFileFromS3(row.key);
+      await this.deleteAttachment(projectId, row.id);
+    }
+  }
+
+  async deleteAllAttachments(projectId: number) {
+    await this.deleteAttachmentsByType(projectId, S3Folder.ATTACHMENTS);
```

Comment:
It might be better to have an actual deleteAllAttachments function, that doesn't require a type.  Otherwise this isn't really a delete all function, its really just a delete whatever types we remember to include function.

Reply:
resolved